src/main/java/Dobby.java

```diff
@@ -0,0 +1,138 @@
+import java.util.Scanner;
+
+public class Dobby {
+
+    private static final String DASH_LINE = "____________________________________________________________";
+    private static final int MAX_LIST_SIZE = 100;
+    private static final Task[] taskList = new Task[MAX_LIST_SIZE];
+
+    private static int listSize = 0;
+
+    public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        String line;
+
+        printWelcomeMessage();
+
+        while (true){
+            line = in.nextLine().trim();
+            if (line.equalsIgnoreCase("bye")){
+                break;
+            }
+            processCommand(line);
+        }
+
+        printGoodbyeMessage();
+    }
+
+    private static void processCommand(String line) {
+        if (line.equals("list")) {
+            printList();
+        } else if (line.startsWith("mark ")) {
+            markTaskAsDone(line);
+        } else if (line.startsWith("unmark ")) {
+            unmarkTaskAsDone(line);
+        } else {
+            addTask(line);
+        }
+    }
+
+    private static void printWelcomeMessage(){
+        printSeparator();
+        System.out.println("    " + "Hello! Dobby is Dobby!");
+        System.out.println("    " + "What can Dobby do for master?");
+        printSeparator();
+    }
+
+    private static void printList(){
+        printSeparator();
+        System.out.println("    Here are the tasks in master's list:");
+        for (int i = 1; i <= listSize; i++) {
+            Task t = taskList[i-1];
+            System.out.println("    " + i + "." + t);
+        }
+        printSeparator();
+    }
+
+    private static void addTask(String line){
+        Task task = createTask(line);
+        addTaskToList(task);
+        printTaskAddedMessage();
+    }
+
+    private static Task createTask(String line){
+        if (line.startsWith("todo ")) {
+            return new Todo(line.substring("todo ".length()));
+        } else if (line.startsWith("deadline ")) {
+            return createDeadlineTask(line);
+        } else if (line.startsWith("event ")){
+            return createEventTask(line);
+        }
+        return new Task(line);
+    }
+
+    private static Deadline createDeadlineTask(String line){
```

Comment:
There is a missing white space for the method
```suggestion
    private static Deadline createDeadlineTask(String line) {
```